### DISCUSSION TITLE
Is it planned to support sending of multiple assets which have been minted in the same on-chain transaction?

### DISCUSSION THREAD

**Discussion by rkedez** [lightninglabs/taproot-assets#916] [General]
The use case would be to mint many different assets (e.g. 100k collectibles) in one on-chain transaction (or a few) and then send all of these assets or a large part of them (e.g. 80k) in one on-chain transaction. Is it planned to support this use case? If so, is there an ETA?

As far as I understand it, this is currently not possible. After PR #821 I have to tried to use the vPSBT workflow to test this with two collectibles minted in one batch/on-chain tx but `FundVirtualPsbt` of the 2nd asset leads to `error funding address send: unable to list eligible coins: failed to find coin(s) that satisfy given constraints; if previous transfers are un-confirmed, wait for them to confirm before trying again`. On [Slack](https://lightningcommunity.slack.com/archives/C03B3556HQ8/p1715611660051529) @guggero answered: “[This] is expected. If you have multiple assets/batches in the same on-chain transaction and spend one of them, then you need to wait for a confirmation until you can spend the change.”

I had once opened issue #670 which was closed with #821 but in this issue I had only mentioned the send of many unique assets which should be possible now. However, it is not feasible to mint all assets in different on-chain transactions in order to be able to send them later with only one on-chain transaction.

**Comment by jharveyb** [comment]
This should already be possible; you could mint the 100k, and then send 80k at once (using the vPSBT workflow).

The earlier feedback was that, while the transfer of those 80k assets is unconfirmed, you wouldn't be able to transfer the 20k other assets from the initial mint. This is because those 20k are being moved to a change output, but until it's confirmed we won't consider those assets during coin selection.
